### PR TITLE
Move uploading nightly wheels to different job

### PR DESCRIPTION
**Pull request by lysnikolaou**

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -269,6 +255,23 @@ jobs:
         name: dist-sdist
         path: dist/*.tar.gz
 
+  scientific-python-nightly-wheels-publish:
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    needs: [build-2-native-wheels, windows]
+    runs-on: ubuntu-latest
+    name: Upload release to PyPI
```
Comment (nulano) [inline .github/workflows/wheels.yml:262]:
```suggestion
    name: Upload release to scientific-python-nightly-wheels
```
?

Comment (radarhere) [inline .github/workflows/wheels.yml:262]:
```suggestion
    name: Upload wheels to scientific-python-nightly-wheels
```

Comment (lysnikolaou) [reply .github/workflows/wheels.yml]: Thanks for the suggestion @nulano. I took the other one since it was fixing the "release" -> "wheels" issue as well.

**Review by radarhere — APPROVED** [review]

**Comment by lysnikolaou** [comment]
Thanks for the review and merging folks! Could we rerun the wheels job to get windows/macos wheels uploaded as well?

**Comment by hugovk** [comment]
There we go: https://github.com/python-pillow/Pillow/actions/runs/9990858119

**Comment by lysnikolaou** [comment]
Yay, we've got windows/linux/macos nightlies on https://anaconda.org/scientific-python-nightly-wheels/pandas/files.

**Comment by aclark4life** [comment]
> Yay, we've got windows/linux/macos nightlies on https://anaconda.org/scientific-python-nightly-wheels/pandas/files.

Yay! You mean https://anaconda.org/scientific-python-nightly-wheels/pillow/files I assume 😄 Thanks all

**Pull request merged**